resources/js/Shared/Components/Panel.vue

```diff
@@ -0,0 +1,61 @@
+<script setup>
+import Map from '@/Shared/Components/Map.vue'
+import { ref, onMounted, onBeforeUnmount, watch } from 'vue'
+import Info from '@/Shared/Components/Info.vue'
+import SearchPanel from '@/Shared/Components/SearchPanel.vue'
+import Nav from '@/Shared/Layout/Nav.vue'
+import { breakpointsTailwind, useBreakpoints } from '@vueuse/core'
+import { XMarkIcon, MapIcon } from '@heroicons/vue/24/outline'
+
+const breakpoints = useBreakpoints(breakpointsTailwind)
+const showInfo = ref(true)
+const isMobile = ref(breakpoints.smaller('lg'))
+const showMapMobile = ref(false)
+
+function switchPanel() {
+  showInfo.value = !showInfo.value
+}
+
+function switchMap() {
+  showMapMobile.value = !showMapMobile.value
+}
+
+onMounted(() => {
+  window.addEventListener('resize', updateIsMobile)
+})
+onBeforeUnmount(() => {
+  window.removeEventListener('resize', updateIsMobile)
+})
+
+function updateIsMobile() {
+  if (!isMobile.value) {
+    showMapMobile.value = false
+  }
+}
+
+watch(() => isMobile, updateIsMobile)
```

Comment:
I think you can did it better. Maybe we don't need onMounted and onBeforeUnmount

Reply:
How would you do it? It watches for window resizings and works fine. The only thing i would change is to remove listener in onUnmounted hook instead of onBeforeUnmount.

Reply:
I think it's okay now.